### PR TITLE
Load ChatListItem details when only ID provided

### PR DESCRIPTION
**Pull request by plasne**
This PR allows for the chat prop sent to ChatListItem to only include the id (as shown below). If that is the case, the ChatListItem will load the other details it needs (title, membership, type, etc.).

```
<ChatListItem key={c.id} chat={{ id: c.id }} myId={chatState.userId} onSelected={props.onSelected} />
```

There are a few other changes as well:
- If there is no valid timestamp (for instance, it hasn't loaded yet), no error is displayed.
- If there is no valid title, the id is used so at least the user can disambiguate between chat threads.

The use of `chatInternal` is noteworthy. If we simply populated the chat details then the component does not re-render. Since chat isn't stored in state at all, passing a setState prop is also not appropriate. Therefore, the ChatListItem component itself creates a `chatInternal` object that is managed by state. It can be changed inside the component (for example, on loading) and it can be reset whenever the chat is changed externally.

### CODE REVIEW

File: packages/mgt-chat/src/components/ChatList/ChatList.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ export const ChatList = (props: MgtTemplateProps & IChatListItemInteractionProps
     <FluentThemeProvider fluentTheme={FluentTheme}>
       <FluentProvider theme={webLightTheme}>
         {chats.map(c => (
-          <ChatListItem key={c.id} chat={c} myId={chatState.userId} onSelected={props.onSelected} />
+          <ChatListItem key={c.id} chat={{ id: c.id }} myId={chatState.userId} onSelected={props.onSelected} />
```
Comment (seekdavidlee) [inline packages/mgt-chat/src/components/ChatList/ChatList.tsx]: I may be missing this but where is the code to set chatInternal from externally?

File: packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx [hunk 2]
```diff
@@ -195,19 +202,41 @@ export const ChatListItem = ({ chat, myId, onSelected }: IMgtChatListItemProps &
...
+    setChatInternal(chat);
+  }, [chat]);
+
+  // enrich the chat if necessary
+  useEffect(() => {
+    if (chatInternal.id && (!chatInternal.chatType || !chatInternal.members)) {
+      const provider = Providers.globalProvider;
+      if (provider && provider.state === ProviderState.SignedIn) {
+        const graph = provider.graph.forComponent('ChatListItem');
+        const load = async (id: string): Promise<Chat> => {
```
Comment (seekdavidlee) [inline packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx]: Seemed like this code can be simplified.

```ts
const load = (id: string): Promise<Chat> => {
  return loadChatWithPreview(graph, id);
};
```

File: packages/mgt-chat/src/components/ChatList/ChatList.tsx (hunk 1 above)
Comment (plasne) [reply packages/mgt-chat/src/components/ChatList/ChatList.tsx]: You don't - you set chat externally. Inside the component, it sets chatInternal to chat and can load additional properties if needed.

File: packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx (hunk 2 above)
Comment (plasne) [reply packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx]: yes, I think you are right.

Comment (seekdavidlee) [reply packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx:216]: I was thinking the Chat state for ChatListItem should be managed externally by ChatList. I understand there is the idea that each component should be responsible for its own state. But if we consider ChatListItem to be an internal component and only used in context of ChatList, wouldn't be safe for its state to be managed by ChatList?

File: packages/mgt-chat/src/components/ChatList/ChatList.tsx (hunk 1 above)
Comment (seekdavidlee) [reply packages/mgt-chat/src/components/ChatList/ChatList.tsx]: I was thinking the Chat state for ChatListItem should be managed externally by ChatList. I understand there is the idea that each component should be responsible for its own state. But if we consider ChatListItem to be an internal component and only used in context of ChatList, wouldn't be safe for its state to be managed by ChatList?

Comment (kavithasreenath) [inline packages/mgt-chat/src/components/ChatList/ChatList.tsx]: Please check if the comments in the ChatList function are still relevant

File: packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx [hunk 3]
```diff
@@ -102,12 +106,12 @@ export const ChatListItem = ({ chat, myId, onSelected }: IMgtChatListItemProps &
         ? `${other?.displayName || (other as AadUserConversationMember)?.email || other?.id}`
         : `${me?.displayName} (You)`;
     }
-    return chatObj.topic || chatObj.chatType;
+    return chatObj.topic || chatObj.chatType || chatObj.id;
```
Comment (kavithasreenath) [inline packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx]: Should we take out chatType?

File: packages/mgt-chat/src/components/ChatList/ChatList.tsx (hunk 1 above)
Comment (plasne) [reply packages/mgt-chat/src/components/ChatList/ChatList.tsx]: ChatListItem being a "sub-component" of ChatList would make it safe to share state. However, there is really no reason to share state, using chatInternal seems to work well and is similar to personDetailsInternal which is already used in MGT.

Comment (plasne) [reply packages/mgt-chat/src/components/ChatList/ChatList.tsx]: I just re-read them; I don't think any of that changes.

File: packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx (hunk 3 above)
Comment (plasne) [reply packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx]: yes, will do.

**Review by kavithasreenath — APPROVED** [review]

**Review by seekdavidlee — APPROVED** [review]

File: packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx [hunk 4]
```diff
@@ -195,19 +202,41 @@ export const ChatListItem = ({ chat, myId, onSelected }: IMgtChatListItemProps &
...
       onClick={() => {
-        onSelected(chat);
+        onSelected(chatInternal);
       }}
     >
       <div className={styles.profileImage}>{getDefaultProfileImage()}</div>
       <div className={styles.chatInfo}>
-        <h3 className={styles.chatTitle}>{inferTitle(chat)}</h3>
-        <p className={styles.chatMessage}>{enrichPreviewMessage(chat.lastMessagePreview)}</p>
+        <h3 className={styles.chatTitle}>{inferTitle(chatInternal)}</h3>
```
Comment (TechnicallyWilliams) [inline packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx]: After merging latest, when the Chat is changed, it resets and that temporary reset state can be seen visually until the load is complete. Should we address that now? 
image

Comment (plasne) [reply packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx]: That was intentional; that way it is still possible to disambiguate between chat threads.

**Pull request merged**